### DISCUSSION TITLE
Export Data as JSON (to avoid lock-in)

### DISCUSSION THREAD

**Discussion by lamyergeier** [TagStudioDev/TagStudio#266] [Ideas]
Is it possible to export a JSON file that contains 

## array of objects with file path and associated tag array

```json
[
	{
		"filepath": "/mnt/a/b.jpg",
		"tag": [
			"a1",
			"b2",
			"c1"
		]
	}
]
```

## array of objects with tags and associated array of filenames 

```json
[
	{
		"tag": "happy",
		"filepath": [
			"/mnt/a1.md",
			"/mnt/b2.jpg",
			"/mnt/c1.pdf"
		]
	}
]
```

**Comment by CyanVoxel** [comment]
This is already planned and tracked with the open issues #47 and #196. We're currently restructuring the backend database but once that's complete we'll be working on the ability to export the library in non-referential human readable format similar to what you're suggesting.

The last thing we want is to have people's library data locked-in to TagStudio, even if the program itself is open source. In addition we'd like an easy way to export and share tags and other things between users, so this fits in with both of those goals.
